lifecycle-operator/controllers/lifecycle/keptntask/job_utils_test.go

```diff
@@ -201,6 +201,66 @@ func TestKeptnTaskReconciler_updateTaskStatus(t *testing.T) {
 	require.Equal(t, apicommon.StateSucceeded, task.Status.Status)
 }
 
+func TestKeptnTaskReconciler_generateJob(t *testing.T) {
+	namespace := "default"
+	taskName := "my-task"
+	svcAccname := "svcAccname"
+	taskDefinitionName := "my-task-definition"
+	token := true
+
+	taskDefinition := makeTaskDefinitionWithServiceAccount(taskDefinitionName, namespace, svcAccname, &token)
+	taskDefinition.Spec.ServiceAccount.Name = svcAccname
+	fakeClient := fakeclient.NewClient(taskDefinition)
+	task := makeTask(taskName, namespace, taskDefinitionName)
+
+	r := &KeptnTaskReconciler{
+		Client:      fakeClient,
+		EventSender: controllercommon.NewK8sSender(record.NewFakeRecorder(100)),
+		Log:         ctrl.Log.WithName("task-controller"),
+		Scheme:      fakeClient.Scheme(),
+	}
+
+	err := fakeClient.Create(context.TODO(), task)
+	require.Nil(t, err)
+
+	ctx := context.TODO()
+	request := ctrl.Request{
+		NamespacedName: types.NamespacedName{
+			Namespace: namespace,
+		},
+	}
+
+	errTask := fakeClient.Get(context.TODO(), types.NamespacedName{
+		Namespace: namespace,
+		Name:      task.Name,
+	}, task)
+	require.Nil(t, errTask)
+
+	errTaskDefinition := fakeClient.Get(context.TODO(), types.NamespacedName{
+		Namespace: namespace,
+		Name:      taskDefinition.Name,
+	}, taskDefinition)
+	require.Nil(t, errTaskDefinition)
+
+	resultingJob, err := r.generateJob(ctx, task, taskDefinition, request)
+	require.Nil(t, err)
+	require.NotNil(t, resultingJob, "generateJob function return a valid Job")
+
+	require.Len(t, resultingJob.Spec.Template.Spec.Containers, 1)
```

Comment:
let's use `require.Equal()` here

Reply:
I think we shd remove the container, since we aren't creating it either.

Reply:
we can check that it is empty